### PR TITLE
Refactor code to take reentrant flags out of loops

### LINKED ISSUE #362: Reentrant flag in CON_PrimaryAffls_TDTM needs to be moved out of loop
Reentrant flags are set inside the for loop that starts in line 74:

https://github.com/SalesforceFoundation/HEDAP/blob/b448807f63a7a2f7ecef454ee39358a1c8f7453a/src/classes/CON_PrimaryAffls_TDTM.cls#L93

https://github.com/SalesforceFoundation/HEDAP/blob/b448807f63a7a2f7ecef454ee39358a1c8f7453a/src/classes/CON_PrimaryAffls_TDTM.cls#L133

We should check for the right action first, then loop through the records.

### PR DESCRIPTION
**Pull request by davidjray**
# Critical Changes

# Changes

# Issues Closed
Fixes #362

### CODE REVIEW

File: src/classes/CON_PrimaryAffls_TDTM.cls [hunk 1]
```diff
@@ -69,68 +69,72 @@ public class CON_PrimaryAffls_TDTM extends TDTM_Runnable {
...
+                        if(newContact.get(afflMapper.primaryAfflFieldNames[j]) != null) {
+                            UTIL_Debug.debug('****Contact has been created with one of the Key Affiliation fields populated');
+                            createAfflFromLookup(newContact, j, afflsToInsert);                           
                         }
+                    } catch(Exception e) {
+                        UTIL_Debug.debug('****Exception message: ' + e.getMessage());
+                        UTIL_Debug.debug('****Stack trace: \n' + e.getStackTraceString());
+                        newContact.addError(e.getMessage());
                     }
-                    alreadyRunAfterInsert = true;
```
Comment (ceiroa) [inline src/classes/CON_PrimaryAffls_TDTM.cls]: We still want to have re-entrant flags, just not inside the for loop. That is, we want all records for the transaction to be processed, but the method to process them no more than once in the same transaction.

File: src/classes/CON_PrimaryAffls_TDTM.cls [hunk 2]
```diff
@@ -69,68 +69,72 @@ public class CON_PrimaryAffls_TDTM extends TDTM_Runnable {
...
+                            UTIL_Debug.debug('****Key Affiliation field has been cleared');
+                            populateConToAccsMap(newContact, oldContact, changedConToAccsMap, j);
                         }
+                        
+                    } catch(Exception e) {
+                        UTIL_Debug.debug('****Exception message: ' + e.getMessage());
+                        UTIL_Debug.debug('****Stack trace: \n' + e.getStackTraceString());
+                        newContact.addError(e.getMessage());
                     }
-                    alreadyRunBeforeUpdate = true;
```
Comment (ceiroa) [inline src/classes/CON_PrimaryAffls_TDTM.cls]: Same here. We want to keep the re-entrant flag.

File: src/classes/CON_PrimaryAffls_TDTM.cls (hunk 1 above)
Comment (davidjray) [inline src/classes/CON_PrimaryAffls_TDTM.cls]: Done.

File: src/classes/CON_PrimaryAffls_TDTM.cls (hunk 2 above)
Comment (davidjray) [inline src/classes/CON_PrimaryAffls_TDTM.cls]: Done

**Comment by davidjray** [comment]
@ceiroa Fixed.

File: src/classes/CON_PrimaryAffls_TDTM.cls [hunk 3]
```diff
@@ -71,7 +71,7 @@ public class CON_PrimaryAffls_TDTM extends TDTM_Runnable {
         Map<ID, List<ID>> changedConToAccsMap = new Map<ID, List<ID>>();
 
         // AFTER INSERT
-        if( newlist != null && triggerAction == TDTM_Runnable.Action.AfterInsert ) {
```
Comment (davidjray) [inline src/classes/CON_PrimaryAffls_TDTM.cls]: Got them back in there @ceiroa

**Pull request merged**